frontend/packages/dev-console/src/components/topology/list-view/TopologyListView.tsx

```diff
@@ -259,40 +270,46 @@ const ConnectedTopologyListView: React.FC<TopologyListViewProps &
       // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [namespace, updateMetrics, updateMonitoringAlerts]);
 
-    if (!visualizationReady) {
-      return null;
-    }
+    const listView = React.useMemo(() => {
```

Comment:
Did you want to use use `React.memo` instead and break this into a pure component?